ethtxmanager/ethtxmanager.go

```diff
@@ -63,13 +63,25 @@ func (c *Client) TrackSequenceBatchesSending(ctx context.Context) {
 	for {
 		select {
 		case sequences := <-c.sequencesToSendChan:
-			err := c.sequenceBatches(sequences)
+			gas, err := c.ethMan.EstimateGasSequenceBatches(sequences)
 			var attempts uint32
 			for err != nil && attempts < c.cfg.MaxSendBatchTxRetries {
-				log.Errorf("failed to sequence batches, trying once again, retry #%d, err: %v", attempts, err)
+				log.Errorf("failed to estimate gas for sending sequences batches, retry #%d, err: %v", err)
 				time.Sleep(c.cfg.FrequencyForResendingFailedSendBatches.Duration)
 				attempts++
-				err = c.sequenceBatches(sequences)
+				gas, err = c.ethMan.EstimateGasSequenceBatches(sequences)
+			}
+
+			gasLimit := uint64(float64(gas) * gasLimitIncrease)
+			attempts = 0
+			err = c.sequenceBatches(sequences, gasLimit)
+			for err != nil && attempts < c.cfg.MaxSendBatchTxRetries {
+				log.Errorf("failed to sequence batches, trying once again, retry #%d, gasLimit: %d, err: %v",
+					attempts, gasLimit, err)
+				time.Sleep(c.cfg.FrequencyForResendingFailedSendBatches.Duration)
+				attempts++
+				gasLimit = uint64(float64(gasLimit) * gasLimitIncrease)
```

Comment:
You should increase the GasPrice instead of the gas. 
As you may know gas is the effort unit to process the tx and gasprice is the amount of wei you are willing to pay for each gas unit.
You should be able to tweak the value here `auth.GasPrice`

Reply:
Resolving conversation as the affected code by the review has been completely removed